project/LagomPublish.scala

```diff
@@ -15,18 +16,22 @@ object LagomPublish {
       case ("lagom-sbt-plugin", x) =>
         // see https://github.com/sbt/sbt-bintray/blob/7c93bacaae3ffc128564ceacb6e73ec4486525dd/src/main/scala/Bintray.scala#L16-L29 for
         // details on the syntax of Bintray Resolver names.
-        if (x.get.name != "Bintray-Sbt-Publish-lagom-sbt-plugin-releases-lagom-sbt-plugin") {
-          throw new PublishValidationFailed("""Invalid resolver. Expected: "Raw(Bintray-Sbt-Publish-lagom-sbt-plugin-releases-lagom-sbt-plugin)".""")
+        if (inReleaseVersion && x.get.name != "Bintray-Sbt-Publish-lagom-sbt-plugin-releases-lagom-sbt-plugin") {
+          throw new PublishValidationFailed(s"""Invalid resolver. Expected: "Raw(Bintray-Sbt-Publish-lagom-sbt-plugin-releases-lagom-sbt-plugin)" but was "${x.get.name}".""")
         }
+        // TODO: Add a validation for "lagom-sbt-plugin" when the version is a snapshot.
       case (_, x) =>
+        // TODO: this could be improved to assert the specific Resolver depending on release-vs-snapshot nature of the version.
+        // e.g. sonatype-staging vs sonatype-snapshots
         if (!x.get.name.toLowerCase.contains("sonatype")) {
-          throw new PublishValidationFailed(s"""Invalid resolver. Expected: Sonatype. Actual $x.""")
+          throw new PublishValidationFailed(s"""Invalid resolver. Expected: Sonatype. Actual ${x.get}.""")
```

Comment:
I don't get why here we use `x.get`, but on the plugin one we use `x.get.name`. What am I missing?

Reply:
You're right. I'll overload the Exception constructor and simplify further.